frame/support/src/storage/types/counted_map.rs

```diff
@@ -262,9 +263,10 @@ where
 	}
 
 	/// Remove all value of the storage.
-	pub fn remove_all() {
-		CounterFor::<Prefix>::set(0u32);
-		<Self as MapWrapper>::Map::remove_all(None);
+	pub fn remove_all(maybe_limit: Option<u32>) {
+		let leftover = Self::count().saturating_sub(maybe_limit.unwrap_or_else(Bounded::max_value));
```

Comment:
this is the only logical change here.

Reply:
this is not correct, the documentation of remove_all for the storage map is indeed missing, but underneath it calls [sp-io::storage::clear_prefix](https://paritytech.github.io/substrate/master/sp_io/storage/fn.clear_prefix.html)
Which deletes all keys in the overlay and up to limit in the backend.

Thus we cannot keep an accurate leftover here, we need to change sp-io::storage::clear_prefix to return here the number of key deleted in the overlay.

Reply:
@kianenigma wat happens nao?

Reply:
follow up is https://github.com/paritytech/substrate/pull/10231